src/main/java/domain/GameManger.java

```diff
@@ -0,0 +1,100 @@
+package domain;
+
+import domain.user.Dealer;
+import domain.user.Player;
+import domain.user.User;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class GameManger {
+    private static final int USER_MIN_COUNT = 1;
+    private static final int USER_MAX_COUNT = 7;
+    private static final int INIT_CARD_NUMBER = 2;
+
+    private final List<Player> users;
+    private final Dealer dealer;
+    private final CardDeck cardDeck;
+
+    public GameManger(List<Player> users, Dealer dealer, CardDeck cardDeck) {
+        validate(users);
+        this.users = new ArrayList<Player>(users);
+        this.dealer = dealer;
+        this.cardDeck = cardDeck;
+    }
+
+    private void validate(List<Player> users) {
+        boolean hasDuplicate = users.stream()
+                .map(Player::getName)
+                .collect(Collectors.toSet())
+                .size() < users.size();
+        if (hasDuplicate) {
+            throw new IllegalArgumentException("유저는 중복될 수 없습니다.");
+        }
```

Comment:
지금처럼 체이닝으로 쭉 처리하기 보다는 적절히 변수로 분리해서 처리하면 가독성이 좋아질 것 같아요. 😃

```suggestion
        long distinctUserNames = users.stream()
            .map(Player::getName)
            .distinct()
            .count();

        if (users.size() != distinctUserNames) {
            throw new IllegalArgumentException("유저는 중복될 수 없습니다.");
        }
```